src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2531,9 +2531,7 @@ uint8_t RadioSpinel<InterfaceType, ProcessContextType>::GetCslAccuracy(void)
     LogIfFail("Get CSL Accuracy failed", error);
     return accuracy;
 }
-#endif
 
-#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
 template <typename InterfaceType, typename ProcessContextType>
 uint8_t RadioSpinel<InterfaceType, ProcessContextType>::GetCslUncertainty(void)
```

Comment:
Is there any possibility of this change creating backward incompatibility between host and RCP?

If so, do we need to handle it ?

I assume we removed this `#if` check here since the new host code will use `GetCslUncertainty()` under `OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE` but then it won't be provided by an RCP running older version?

Reply:
`SPINEL_PROP_RCP_CSL_UNCERTAINTY` is already part of the RCP since it was needed for `OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE` builds.

My understanding is that we are only modifying the host side here, right? It shouldn't be an issue.

Reply:
Wondering in what cases would the host side of a CSL receiver request the CSL uncertainty?  The real usage of it is on the radio (RCP) side in the sub-MAC. If never requested, we could leave the `#if` here in place. 

I'm assuming here there could be a legacy "CSL Receiver only" build of the RCP. (Assumption from @edmont seems that the RCP build includes both CSL Receiver and Transmitter code.)

Reply:
Good points @EskoDijk @abtink. I've updated the PR to keep the host unchanged.